### PR TITLE
Replace Variable PROJECT_NAME in CMakeLists.txt with the actual name

### LINKED ISSUE #1920: Replace unhelpful cmake variable PROJECT_NAME with actual package name
image

`${PROJECT_NAME}` is a variable that is found in many of our cmake files. [here is one example](https://github.com/ros-planning/moveit2/blob/2105b91f89ea213a0fc3fc6b3735d296b5d57c3f/moveit_kinematics/CMakeLists.txt#L68). The project name in the linked to example is found at the top of the file `project(moveit_kinematics)`. It would make this file more readable if all instances of `${PROJECT_NAME}` was replaced with `moveit_kinematics`.

Tasks:
- [ ] Find `${PROJECT_NAME}` in cmake files
- [ ] Replace with the project name.
- [ ] Be Happy!

### PR DESCRIPTION
**Pull request by Shobuj-Paul**
### Description

Replaced PROJECT_NAME variable with the contents of the project(*) in CMakeLists.txt of various packages.
Fixes #1920.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by sjahr — APPROVED** [review]

**Review by ChrisThrasher — APPROVED** [review]

**Comment by Shobuj-Paul** [comment]
> Thanks, @Shobuj-Paul, I opened a PR to address the compiler error in CI and then we're good to merge this PR!

Cool thanks, I had opened an issue to address the same. I think it would be safe to close it.

File: moveit_ros/moveit_servo/CMakeLists.txt [hunk 1]
```diff
@@ -43,6 +43,7 @@ add_library(moveit_servo_lib_parameters SHARED
   src/servo_parameters.cpp
   src/parameter_descriptor_builder.cpp
 )
+
```
Comment (sjahr) [inline moveit_ros/moveit_servo/CMakeLists.txt:46]:
```suggestion
```

File: moveit_ros/moveit_servo/CMakeLists.txt [hunk 2]
```diff
@@ -54,6 +55,7 @@ add_library(moveit_servo_lib SHARED
   src/servo_calcs.cpp
   src/utilities.cpp
 )
+
```
Comment (sjahr) [inline moveit_ros/moveit_servo/CMakeLists.txt:58]:
```suggestion
```

**Comment by Shobuj-Paul** [comment]
Should I update this branch with a merge commit? Rebase has some conflicts.

**Comment by sjahr** [comment]
> Should I update this branch with a merge commit? Rebase has some conflicts.

@Shobuj-Paul No :) Looks good now, no need for you to do anything. I'll ask people for a second review and then we can merge it

**Review by tylerjw — APPROVED** [review]

**Pull request merged**